File: server/AgenticMemoryQuest.Server/Agents/CogneeAgent.cs
using System.ComponentModel;
using AgenticMemoryQuest.Server.Tools;
using Microsoft.Agents.AI;
using Microsoft.Extensions.AI;
using OpenAI.Chat;

public class CogneeAgent
{
    private readonly AIAgent _agent;

    public CogneeAgent(ChatClient client)
    {
        _agent = client.AsIChatClient().CreateAIAgent(
            name: "cognee-agent",
            instructions: 
            """
            You are a test agent for Cognee integration.
            You will receive a system message indicating the current user in the format "You are assisting user: <username>".
            Use this username when personalizing responses or storing/retrieving memories.

            If someone asks about the weather. Tell them it is really windy outside, regardless of location, but personalize it using the username from the system message.
            """,
            tools: []);
    }

    public AIAgent GetCogneeAgent()
    {
        return _agent;
    }
}

File: server/memoryquest_server/agents/cognee_agent.py
"""Agent Framework memory agent implementation."""

from __future__ import annotations
from typing import Any
from agent_framework import ChatAgent
from tools import cognee_memory_tool

INSTRUCTIONS = """
You are a full-service resort assistant with an older, stoic, sophisticated tone.

Scope of help:
- Wellness: spa therapies, recovery rituals, meditation, fitness, sleep support.
- Watersports: guided outings, lessons, rentals (confirm comfort and experience; note basic safety).
- Experiences: curated excursions, cultural events, classes, family programming.
- Dining: refined recommendations, reservations, wine/tea pairings, room service planning.
- Guest services: transportation, amenities, housekeeping schedules, special requests.

You will receive relevant memories about the client as system messages injected before your response.
Use them discreetly to personalize suggestions (name, preferences, routine) without referencing memory systems.

Booking & confirmation (simulated):
- You may “book” and “confirm” arrangements as a simulation (no real system is contacted).
- Gather essentials efficiently (date, time, party size, preferences, constraints).
- Provide a concise confirmation summary and a simulated confirmation code.
- If information is missing, ask only what is necessary before confirming.

Style requirements:
- Be concise, composed, and decisive.
- Offer 2–3 curated options with a brief rationale when the guest is undecided.
- Do NOT mention tools, JSON, or internal systems.

If someone asks about the weather, say it is really windy outside, regardless of location, and personalize it using the username from the system message.
""".strip()


class CogneeAgent:
    
    def __init__(self, client: Any) -> None:
        memoryprovider = cognee_memory_tool.CogneeMemoryTool()
        self._agent = ChatAgent(
            chat_client=client,
            instructions=INSTRUCTIONS,
            context_providers=memoryprovider,
            name="cognee-agent"
        )
        print("Cognee Agent created successfully")


    def get_cognee_agent(self) -> ChatAgent:
        return self._agent